.github/workflows/unit_test.yml

```
@@ -0,0 +1,86 @@
+name: Unit Tests
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    branches-ignore:
+      - master
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+env:
+  DEFAULT_PYTHON: '3.11'
+
+jobs:
+  tests:
+    name: Run unit tests (Python 3.11, Pydantic v1 on ubuntu-latest)
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    env:
+      POETRY_VERSION: 1.6.1
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.6.1
+
+      - name: Cache poetry
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pypoetry
+          key: python-311-codeql-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: |
+            python-311-codeql-${{ hashFiles('**/poetry.lock') }}
```

```suggestion
          key: ${{ runner.os }}-python-${{ env.DEFAULT_PYTHON }}-tests-unit-${{ hashFiles('**/poetry.lock') }}
          restore-keys: |
            ${{ runner.os }}-python-${{ env.DEFAULT_PYTHON }}-tests-unit-${{ hashFiles('**/poetry.lock') }}
            ${{ runner.os }}-python-${{ env.DEFAULT_PYTHON }}-tests-unit-
            ${{ runner.os }}-python
            ${{ runner.os }}-
```
`key` и первый `restore-keys` используются, чтобы зависимости разных тестов не пересекались.

Остальные `restore-keys` испоьзуются, чтобы если кэша с таким key нет, выполнялся fallback на кэш от предыдущих запусков или от других workflow